### PR TITLE
fix(a11y): goal chart bugs

### PR DESCRIPTION
**Pull request by rshen91**
## Summary
This PR addresses some bugs with #1161. 

## Details

- Fixes the minValue is the current value in the goal semantic description
- Fixes key react red error
- Fixes the typo $ under Target for Types.tsx
- Adds `mergePartial(defaultConfig, partialConfig, { mergeOptionalPartialValues: true });` where relevant for RecursivePartial configs (goal charts, heatmap charts, partition charts, word cloud charts)
- I deleted the existing vrts and re-ran them (to account for the number of changed files)

### CODE REVIEW

File: packages/charts/src/components/accessibility/goal_semantic_description.tsx [hunk 1]
```diff
@@ -22,20 +22,20 @@ export const GoalSemanticDescription = ({
...
-            <dt key={`dt--${index}`}>{`${firstValue} - ${value}`}</dt>
-            <dd key={`dd--${index}`}>{`${text[index]}`}</dd>
-          </>
+          <Fragment key={`dtdd--${value}--${text[index]}`}>
+            <dt>{`${prevValue.value} - ${value}`}</dt>
+            <dd>{`${text[index]}`}</dd>
+          </Fragment>
+        ) : firstValue === value ? null : (
+          <Fragment key={`dtdd--${value}--${text[index]}`}>
+            <dt>{`${firstValue} - ${value}`}</dt>
```
Comment (myasonik) [inline packages/charts/src/components/accessibility/goal_semantic_description.tsx:36]: Looks like the only difference in the two return statements is the `firstValue`. 

I wonder if the code will be more readable with a single return statement with the variable computed before hand.

Something like:
```
if (firstValue === value) return; 
const prevValue = bandLabels[index - 1];
const firstValue = prevValue !== undefined ? prevValue.value : firstValue;
return (<Fragment key={`dtdd--${value}--${text[index]}`}>
            <dt>{`${firstValue} - ${value}`}</dt>
            <dd>{`${text[index]}`}</dd>
          </Fragment>);
```

Comment (rshen91) [reply packages/charts/src/components/accessibility/goal_semantic_description.tsx]: Ok nice - way cleaner! 7483ac2 thank you!

Comment (rshen91) [reply packages/charts/src/components/accessibility/goal_semantic_description.tsx]: Oh nice!! Super clean - please see: 7483ac2

**Comment by rshen91** [comment]
jenkins test this

**Review by myasonik — APPROVED** [review]
🚀

**Review by nickofthyme — APPROVED** [review]
LGTM!

**Comment by rshen91** [comment]
jenkins test this

**Comment by rshen91** [comment]
jenkins test this

**Pull request merged**